pkg/azureconditions/deploymenthandler.go

```diff
@@ -20,16 +22,47 @@ const (
 	DeploymentProvisioningStateFailed        = "Failed"
 )
 
-func (r *Resource) checkIfDeploymentIsSuccessful(ctx context.Context, deploymentsClient *resources.DeploymentsClient, cr conditions.CR, deploymentName string, conditionType capi.ConditionType) (bool, error) {
+type DeploymentCheckerConfig struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+// Resource ensures that AzureMachinePool Status Conditions are set.
+type DeploymentChecker struct {
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+func NewDeploymentChecker(config DeploymentCheckerConfig) (*DeploymentChecker, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &DeploymentChecker{
+		ctrlClient: config.CtrlClient,
+		logger:     config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *DeploymentChecker) logWarning(ctx context.Context, message string, messageArgs ...interface{}) {
```

Comment:
wouldn't `logger.Errorf` log error, and not warning?

Reply:
Yes, but after short discussion here: https://github.com/giantswarm/micrologger/pull/60#discussion_r532725012 we figured that it's not necessary. We don't really do anything with the log levels besides color them w/ `luigi` so maybe we can go towards those two and see how it goes.

Reply:
I see, in that case I would probably use `Debugf`, since `Errorf` would trigger alerts. For me personally the use case for warning logs is to look for them while debugging and digging through logs, as it's easier to spot issues.